client/src/app/site/tags/models/view-tag.ts

```diff
@@ -0,0 +1,51 @@
+import { Tag } from '../../../shared/models/core/tag';
+import { TranslateService } from '@ngx-translate/core';
+import { BaseViewModel } from '../../base/base-view-model';
+
+/**
+ * Category class for the View
+ *
+ * Stores a Category including all (implicit) references
+ * Provides "safe" access to variables and functions in {@link Category}
+ * @ignore
+ */
+export class ViewTag extends BaseViewModel {
+    private _tag: Tag;
+
+    public constructor(tag: Tag) {
+        super();
+        this._tag = tag;
+    }
+
+    public get tag(): Tag {
+        return this._tag;
+    }
+
+    public set tag(tag: Tag) {
```

Comment:
I suppose a set function is never necessary? You are not supposed to alter tags on your local device but send changed information to the server. For that you usually use new tags (not viewTags) that you create in your repository